codec-dns/src/main/java/io/netty/handler/codec/dns/DohProviders.java

```diff
@@ -0,0 +1,80 @@
+package io.netty.handler.codec.dns;
+
+public class DohProviders {
+
+    public static class DohProvider {
+        private final boolean usePost;
+        private final String host;
+        private final int port;
+        private final String uri;
+
+        public DohProvider(String host) {
+            this(host, "/dns-query", 443);
+        }
+
+        public DohProvider(String host, boolean usePost) {
+            this(host, "/dns-query", 443, usePost);
+        }
+
+        public DohProvider(String host, String uri) {
+            this(host, uri, 443);
+        }
+
+        public DohProvider(String host, String uri, int port) {
+            this(host, uri, port, true);
+        }
+
+        public DohProvider(String host, String uri, int port, boolean usePost) {
+            this.host = host;
+            this.uri = uri;
+            this.port = port;
+            this.usePost = usePost;
+        }
+
+        public boolean usePost() {
+            return usePost;
+        }
+
+        public String host() {
+            return host;
+        }
+
+        public int port() {
+            return port;
+        }
+
+        public String uri() {
+            return uri;
+        }
+    }
+
+    public static DohProvider GOOGLE = new DohProvider("dns.google");
```

Comment:
Are these needed here, or better fit in examples?

Reply:
sure, I will move them to examples.